Add all needed dependencies to fluss-dist

### Search before asking

- [x] I searched in the [issues](https://github.com/alibaba/fluss/issues) and found nothing similar.


### Fluss version

0.5.0 (latest release)

### Please describe the bug 🐞

Currently, `fluss-dist/src/main/assemble/bin.xml`, `fluss-dist/src/main/assemble/plugin.xml`,`fluss-dist/src/main/assemble/opt.xml` packages a bundle of dependencies.

Howerver, some dependencies lacks in pom.xml of fluss-dist. If we not run `mvn clean install -DskipTests` to build all the modules,  but run `mvn clean install -pl fluss-dist -am -DskipTests` to build fluss-dist(and its dependencies) only. Error will occurs:
`[ERROR] Failed to execute goal org.apache.maven.plugins:maven-assembly-plugin:3.6.0:single (plugins) on project fluss-dist: Failed to create assembly: Error adding file to archive:  /fluss/fluss-dist/../fluss-metrics/fluss-metrics-prometheus/target/fluss-metrics-prometheus-0.6-SNAPSHOT.jar`


### Solution

Add all needed dependencies to fluss-dist

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

--- a/fluss-dist/pom.xml
+++ b/fluss-dist/pom.xml
@@ -37,6 +37,14 @@
             <scope>provided</scope>
         </dependency>
 
+        <!--  filesystem plugin -->
+        <dependency>
+            <groupId>com.alibaba.fluss</groupId>
+            <artifactId>fluss-fs-hadoop</artifactId>
+            <version>${project.version}</version>
+            <scope>provided</scope>
+        </dependency>
+
         <dependency>
             <groupId>com.alibaba.fluss</groupId>
             <artifactId>fluss-fs-oss</artifactId>
@@ -51,6 +59,22 @@
             <scope>provided</scope>
         </dependency>
 
+        <!-- metrics plugin -->
+        <dependency>
+            <groupId>com.alibaba.fluss</groupId>
+            <artifactId>fluss-metrics-prometheus</artifactId>
+            <version>${project.version}</version>
+            <scope>provided</scope>
+        </dependency>
+
+        <dependency>
+            <groupId>com.alibaba.fluss</groupId>
+            <artifactId>fluss-metrics-jmx</artifactId>
+            <version>${project.version}</version>
+            <scope>provided</scope>
+        </dependency>
+
+
         <dependency>
             <groupId>com.alibaba.fluss</groupId>
             <artifactId>fluss-lakehouse-paimon</artifactId>
@@ -64,6 +88,7 @@
             <version>${project.version}</version>
             <scope>provided</scope>
         </dependency>
+
 
         <!-- Concrete logging framework - we add this only here (and not in the
 			root POM) to not tie the projects to one specific framework and make
@@ -93,6 +118,8 @@
             <artifactId>log4j-1.2-api</artifactId>
             <scope>compile</scope>
         </dependency>
+
+
     </dependencies>
 
 

